src/CommunityPatch/Patches/Perks/Cunning/Tactics/TrustedCommandersPatch.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+using HarmonyLib;
+using Helpers;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Map;
+using static System.Reflection.BindingFlags;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches.Perks.Cunning.Tactics {
+
+  public sealed class TrustedCommandersPatch : PatchBase<TrustedCommandersPatch> {
+
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo = typeof(DefaultArmyManagementCalculationModel).GetMethod("CalculateCohesionChange", Public | Instance | DeclaredOnly);
+    private static readonly MethodInfo PatchMethodInfo = typeof(TrustedCommandersPatch).GetMethod(nameof(Postfix), Public | NonPublic | Static | DeclaredOnly);
+
+    public override IEnumerable<MethodBase> GetMethodsChecked() {
+      yield return TargetMethodInfo;
+    }
+
+    private PerkObject _perk;
+
+    private static readonly byte[][] Hashes = {
+      new byte[] {
+        // e1.2.1.226961
+        0xFF, 0x28, 0x27, 0xC0, 0x56, 0xD1, 0x13, 0x01,
+        0x4D, 0x06, 0x90, 0xA4, 0xE3, 0xF7, 0x0A, 0x85,
+        0x00, 0xBB, 0x7A, 0x42, 0xD3, 0x82, 0x21, 0xFD,
+        0x26, 0xA4, 0x9E, 0xB8, 0xA4, 0x22, 0xFB, 0xB9
+      }
+    };
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "6ETg3maz");
+
+    public override bool? IsApplicable(Game game) {
+      if (_perk == null) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo)) return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, 50f,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        _perk.IncrementType
+      );
+      
+      if (Applied) return;
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, postfix: new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void Postfix(ref float __result, Army army, StatExplainer explanation = null) {
+      var perk = ActivePatch._perk;
+      var reduction = CalculatePerkReduction(army, perk);
```

Comment:
So it's multiplicative tapering off towards 100%? So if 1 party qualifies, 50% reduction for entire army, if two 75%, if three parties 87.5%? I feel like a PartyMember bonus should merely change whether each party qualifies for the perk, and the decision to then apply it to the entire army should be based on specific conditions.

My impression is that it would be more balanced to do it like this:
- If the army's leader party qualifies for the perk, apply it to the entire army's cohesion change. This is how TW does it for the 'Assuring Presence' perk which similarly reduces cohesion decay by 20%. 
- Else get the percentage of men in the army who are in parties with the perk, and scale it based on that. So if that's 20% then the final reduction is 50% * 20% = 10%.

Example of Assuring Presence perk (from `CalculateCohesionChangeInternal`):

```CSharp
if (!army.LeaderParty.HasPerk(DefaultPerks.Leadership.AssuringPresence))
        return;
float perkMultiplyFactor = DefaultPerks.Leadership.AssuringPresence.PrimaryBonus * 0.01f;
float cohesionIncrease = Math.Abs(
    cohesionChange.ResultNumber  // negative
        * perkMultiplyFactor);
cohesionChange.Add(cohesionIncrease, DefaultArmyManagementCalculationModel._assuringPresencePerkText);
```

Reply:
I have pushed the changes according to your suggestions, could you review again please?

Reply:
Looks good!